lib/avatar/widgets/avatar_screen.dart

```diff
@@ -13,129 +18,143 @@ class AvatarScreen extends StatefulWidget {
+                  ),
+                  callback: (selected) {
+                    setState(() {
+                      if (user.ownsItem(selected)) {
+                        user.avatar[type] = selected.resource;
+                      }
+                    });
+                  },
+                )),
+      ],
     );
   }
-}
-
-class PropertyCarousel extends StatefulWidget {
-  final String current;
-  final List<String> options;
-  final String caption;
-  final Function(String) callback;
-
-  const PropertyCarousel(
-      {Key key, this.current, this.options, this.caption, this.callback})
-      : super(key: key);
 
-  @override
-  _PropertyCarouselState createState() => _PropertyCarouselState();
+  String _typeToString(String type) {
+    switch (type) {
+      case 'body':
+        return 'Clothing';
+      case 'face':
+        return 'Faces';
+      case 'hair':
+        return 'Hair Styles';
+    }
+  }
 }
 
-class _PropertyCarouselState extends State<PropertyCarousel> {
-  String current;
+class ShopItemView extends StatelessWidget {
+  final _AvatarScreenState parent;
+  final ShopItem shopItem;
 
-  @override
-  void initState() {
-    current = widget.current;
-    super.initState();
-  }
+  const ShopItemView(
+    this.shopItem, {
+    Key key,
+    this.parent,
+  }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return Column(
-      children: <Widget>[
-        Text(widget.caption),
-        Divider(),
-        Wrap(
-          children: widget.options
-              .map((v) => FlatButton(
-                    child: Material(
-                      elevation: current == v ? 2.0 : 0.0,
-                      shape: RoundedRectangleBorder(
-                          borderRadius: BorderRadius.circular(4.0)),
-                      child: Container(
-                        height: 64,
-                        child: AspectRatio(
-                            aspectRatio: 1.0, child: Image.asset(v)),
-                      ),
-                    ),
-                    onPressed: () {
-                      setState(() {
-                        current = v;
-                      });
-                      widget.callback(current);
-                    },
-                  ))
-              .toList(),
-        )
-      ],
+    var user = Provider.of<UserService>(context);
+    return GestureDetector(
+      onLongPress: () {
+        if (!user.ownsItem(shopItem)) {
+          showDialog(
+            context: context,
+            builder: (context) => AlertDialog(
+              title: Text(shopItem.name),
+              content: Column(
+                children: <Widget>[
+                  Image.asset(shopItem.resource),
+                  Text('${shopItem.cost} HC'),
+                  RaisedButton(
+                    child: Text(
+                        user.ownsItem(shopItem) ? 'You own this item' : 'Buy'),
+                    onPressed: (user.hcCount >= shopItem.cost &&
```

Comment:
`user.hcCount >= shopItem.cost` should probably be handled in the `buy` function